community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/commands/predicates/Predicate.scala

```diff
@@ -157,6 +157,7 @@ case class PropertyExists(variable: Expression, propertyKey: KeyToken) extends P
       propertyKey.getOptId(state.query).exists(state.query.relationshipOps.hasProperty(pc.id, _)))
     case IsMap(map) => Some(map(state.query).get(propertyKey.name) != Values.NO_VALUE)
     case null => None
```

Comment:
This should be 
```
case NO_VALUE => None
```
you can remove the `case null` probably just missed when refactored.

Reply:
Fixed in new commit